### ISSUE TITLE
List of project using Meson in Users.md is outdated

### ISSUE THREAD

**Issue by dnicolodi** [mesonbuild/meson#8890]
The `[Chicken Libraries](https://github.com/chicken-libraries)` link is broken.

The text above the list says "All the software on this list is tested for regressions before release, so it's highly recommended that projects add themselves here." however, I don't this this is the case, or the disappearance of this project would have been noticed. This brings the question: is keeping this list worth the trouble?

**Comment by xclaesse** [comment]
I think a subset of them are tested. I would like to get that page replaced by WrapDB: https://mesonbuild.com/Wrapdb-projects.html.

**Comment by tp-m** [comment]
I don't think it should be replaced by the list of wrapdb projects, that's a completely different list IMHO.

Perhaps the Users page should be trimmed down however and mention a small curated selection of a few major/noteworthy projects? (Basically, something a corporate developer can show to management to show that this isn't some exotic build system but something that's been adopted by some big and widely-used projects).

**Comment by xclaesse** [comment]
My goal with WrapDB is to get most of meson projects there, and have CI that can build them all on meson RC. It's mot much more work to get your project in the Users page than submitting a wrap file. The wrap file could gain `description` and `website_url` fields to be used to generate the Users.md page.

**Comment by tp-m** [comment]
But it would then also include projects which aren't actually users, e.g. ffmpeg or libjpeg-turbo, wouldn't it?

Either way, IMHO less is more here however the list is generated.

**Comment by xclaesse** [comment]
> But it would then also include projects which aren't actually users, e.g. ffmpeg or libjpeg-turbo, wouldn't it?

We can easily skip wraps that have a `patch_directory`, or add a note it's not upstream.

But I agree it makes more sense to have a Users page that is more "marketing" oriented, where we list large projects like GNOME instead of small and unknown tiny projects.

**Comment by eli-schwartz** [comment]
At least we could be somewhat selective of the ones we mention, whereby we only mention the ones that have actual users other than the developer. I think that's the real problem with "chicken-libraries"...

**Comment by dnicolodi** [comment]
It is clear that the list is not used for regression testing and that purpose is much better served by the WrapDB. If the projects list is to be kept for "marketing" purposes, it should be a brief list of maybe 10 or so high-visibility projects covering as much of the spectrum as possible (embedded, system level, user level, libraries, applications, ...). Anything else would turn it into a vanity list (see the dead link issue).

**Comment by jon-turney** [comment]
> It is clear that the list is not used for regression testing

In addition to whatever testing is done when a release is made, this list is used to drive https://github.com/jon-turney/meson-corpus-test

> better served by the WrapDB

I think it only makes sense to list things in wrapdb which are capable of being consumed by other projects, which is only a subset of projects which could use meson.

> My goal with WrapDB is to get most of meson projects there, and have CI that can build them all on meson RC

But yeah, this would be a good thing to have.

**Comment by dnicolodi** [comment]
> In addition to whatever testing is done when a release is made, this list is used to drive https://github.com/jon-turney/meson-corpus-test

Is it? Looking at the repository I don't see anything that references the list in the documentation. The Github Actions workflow seems to contain a manually curated list of source repositories with more information that what is available on the documentation page, thus, I don't see any dependency on it.

**Comment by xclaesse** [comment]
> which is only a subset of projects which could use meson.

That's true that it would include only libraries, not applications, good point.

**Comment by eli-schwartz** [comment]
For now I've just delisted this one thing. But this ticket still raises an interesting point.

Should we establish some policy on what to add to this list? What kind of criteria?

**Comment by jon-turney** [comment]
I'd like to invert things, and rather than parsing Users.md to extract a list of projects to test, instead have this page be generated from a list of projects we test.

(This could also resolve the ambiguity where it's unclear if the URL in Users.md should point to the project homepage or repo, and let us only publish to this page (some subset of) the corpus we test.)

Maybe I'll take a look at doing that...

> Should we establish some policy on what to add to this list? What kind of criteria?

Yeah.  I'm not sure what the objective criteria for 'is important enough to be listed here' could be.

**Comment by jon-turney** [comment]
> I'd like to invert things, and rather than parsing Users.md to extract a list of projects to test, instead have this page be generated from a list of projects we test.

See #8934
